testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -422,6 +422,43 @@ public void testTrinoReadsSparkPartitionedTable(StorageFormat storageFormat, int
         onSpark().executeQuery("DROP TABLE " + sparkTableName);
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testPartitionedByNestedFiled()
+    {
+        String baseTableName = "test_trino_nested_field_partition_" + randomNameSuffix();
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+
+        onSpark().executeQuery(format("" +
+                        "CREATE TABLE %s (" +
+                        "  id INT," +
+                        "  parent STRUCT<nested:STRING>)" +
+                        "  USING ICEBERG" +
+                        "  PARTITIONED BY (parent.nested)" +
+                        "  TBLPROPERTIES ('format-version'=2)",
+                sparkTableName));
+        onSpark().executeQuery("INSERT INTO " + sparkTableName + " VALUES (1, named_struct('nested', 'a')), (2, named_struct('nested', 'b'))");
+
+        assertThat(onTrino().executeQuery("SELECT id, parent.nested FROM " + trinoTableName))
+                .containsOnly(row(1, "a"), row(2, "b"));
+
+        assertQueryFailure(() -> onTrino().executeQuery("INSERT INTO " + trinoTableName + " VALUES (2, ROW('b'))"))
+                .hasMessageContaining("Partitioning by nested field is unsupported parent.nested");
+        assertQueryFailure(() -> onTrino().executeQuery("UPDATE " + trinoTableName + " SET id = 2"))
+                .hasMessageContaining("Partitioning by nested field is unsupported parent.nested");
+        assertQueryFailure(() -> onTrino().executeQuery("DELETE FROM " + trinoTableName))
```

Comment:
Out of curiosity: In a previous version of the PR before this last commit, the `DELETE` operation was still working. Why did it get removed?

Reply:
Relates to #15178. My branch status was before 84848cbafb85c4c78f1b48ebff2e4d7d289dccb1